src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs

```diff
@@ -96,5 +96,30 @@ public static RecordType Empty()
         {
             return new KnownRecordType();
         }
+
+        /// <summary>
+        /// Resolve logical names when display names are given.
+        /// </summary>
+        /// <param name="recordToResolve">RecordValue to resolve.</param>
+        /// <returns>RecordValue containing only logical field names.</returns>
+        internal RecordValue ResolveToLogicalNames(RecordValue recordToResolve)
+        {
+            var list = new List<NamedValue>();
+            var rType = recordToResolve.Type;
+
+            foreach (var field in recordToResolve.Fields)
```

Comment:
>ecordToResolve.Fields) [](http://example.com/codeflow?start=35&length=22)

@LucGenetier - I guessing this is the performance issue that you hit? Enumerating all fields gets expensive for a dataverse type;.

Reply:
Maybe there's some optimized  lazy execution way to do this. (with derived RecordType and RecordValues)

Reply:
Let's stick with Luc's change here (to not call ResolveToLogicalNames on Patch's arg1) and we can pursue this optimization in a separate PR -and then reenable calling it.